### PR TITLE
Fix: window mode in fullscreen

### PR DESCRIPTION
**Pull request by KyleUltimate**
When using window mode in fullscreen, you will still have the selection "window" stuff, when you really shouldn't.
Let me know whether when in fullscreen, should you just copy screen, or select region(area)

### CODE REVIEW

File: grimblast/grimblast [hunk 1]
```diff
@@ -148,14 +148,21 @@ elif [ "$SUBJECT" = "output" ] ; then
...
+  SIZE=$(hyprctl monitors -j | jq -r 'map(.width,.height)')
   WORKSPACES="$(hyprctl monitors -j | jq -r 'map(.activeWorkspace.id)')"
-  WINDOWS="$(hyprctl clients -j | jq -r --argjson workspaces "$WORKSPACES" 'map(select([.workspace.id] | inside($workspaces)))' )"
-  GEOM=$(echo "$WINDOWS" | jq -r '.[] | "\(.at[0]),\(.at[1]) \(.size[0])x\(.size[1])"' | slurp)
-  # Check if user exited slurp without selecting the area
-  if [ -z "$GEOM" ]; then
-   exit 1
+  # Check if user is in fullscreen
+  if [ -z "$(hyprctl clients -j | jq -r ".[] |select (.workspace.id == "$(echo $WORKSPACES | jq '.[]')" and .size == $SIZE)")" ] ; then
+    WINDOWS="$(hyprctl clients -j | jq -r --argjson workspaces "$WORKSPACES" 'map(select([.workspace.id] | inside($workspaces)))' )"
```
Comment (fufexan) [inline grimblast/grimblast:155]: These two calls are redundant (and slow). We should instead set a variable
```sh
CLIENTS=$(hyprctl clients -j)
```
and pass it to `jq` with `<<< $CLIENTS`

File: grimblast/grimblast [hunk 2]
```diff
@@ -148,14 +148,21 @@ elif [ "$SUBJECT" = "output" ] ; then
   OUTPUT=$(hyprctl monitors -j | jq -r '.[] | select(.active == "yes")' | jq -r '.name')
   WHAT="$OUTPUT"
 elif [ "$SUBJECT" = "window" ] ; then
+  SIZE=$(hyprctl monitors -j | jq -r 'map(.width,.height)')
   WORKSPACES="$(hyprctl monitors -j | jq -r 'map(.activeWorkspace.id)')"
```
Comment (fufexan) [inline grimblast/grimblast:152]: Same as with clients, this could be `MONITORS=...`.

**Review by fufexan — CHANGES_REQUESTED** [review]
Haven't had a chance to test this yet, but these are the the obvious things I'd change by only taking a glance over the code.

**Comment by KyleUltimate** [comment]
Done!
Small question, what the difference between `echo "$WINDOWS" | jq -r '.[] | "\(.at[0]),\(.at[1]) \(.size[0])x\(.size[1])"' | slurp` this than redirect shell inputing?

**Comment by fufexan** [comment]
Probably no difference, just the way I did it in my scripts.

**Comment by fufexan** [comment]
I just noticed, this broke window selection (at least on multiple monitors).

I will be reverting this and potentially implement a check when there's only one monitor present.

**Pull request merged**